src/common/utxobased/engine/makeUtxoEngine.ts

```diff
@@ -247,10 +247,8 @@ export async function makeUtxoEngine(
 
     async isAddressUsed(address: string): Promise<boolean> {
       const scriptPubkey = walletTools.addressToScriptPubkey(address)
-      const addressData = await processor.fetchAddressByScriptPubkey(
-        scriptPubkey
-      )
-      return addressData?.used === true
+      const addressData = await processor.fetchAddresses(scriptPubkey)
+      return addressData?.used
```

Comment:
My guess is because it returns a Promise it implicitly handles the undefined case.

Reply:
Ah, so because HashBases can return `(T | undefined)[]` we should fix the return type on `fetchAddresses` to `IAddress | undefined`. Also, "Addresses"? Hmm

Reply:
Ah, so because HashBases can return `(T | undefined)[]` we should fix the return type on `fetchAddresses` to `IAddress | undefined`. Also, "Addresses"? Hmm

Reply:
Solution: return `undefined` if `addressData == null`